stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -246,10 +244,8 @@ namespace stan {
       check_symmetric("cholesky_decompose", "A", A);
 
       Eigen::Matrix<double, -1, -1> L_A(value_of_rec(A));
-      Eigen::LLT<Eigen::MatrixXd> L_factor
-        = L_A.selfadjointView<Eigen::Lower>().llt();
+      Eigen::LLT<Eigen::Ref<Eigen::MatrixXd> > L_factor(L_A);
```

Comment:
Is there a reason we're no longer using a selfadjointView of `L_A`?   Have the input requirements changed now that we use all of it?